jenkins/opensearch-dashboards/Jenkinsfile

```diff
@@ -122,16 +122,20 @@ pipeline {
             steps {
                 node('Jenkins-Agent-al2-x64-c54xlarge-Docker-Host') {
                     script {
-                        def VERSION = sh(script: 'echo ${INPUT_MANIFEST} | grep -Po "[0-9.]+?(?=.yml)"', returnStdout: true).trim()
-                        sh "echo VERSION:${VERSION} BUILD_NUMBER:${env.BUILD_NUMBER}"
-                        def URL_x64 = "${PUBLIC_ARTIFACT_URL}/${env.JOB_NAME}/${VERSION}/${env.BUILD_NUMBER}/linux/x64/dist/opensearch-dashboards-${VERSION}-linux-x64.tar.gz"
-                        def URL_arm64 = "${PUBLIC_ARTIFACT_URL}/${env.JOB_NAME}/${VERSION}/${env.BUILD_NUMBER}/linux/arm64/dist/opensearch-dashboards-${VERSION}-linux-arm64.tar.gz"
+                        def inputManifest = lib.InputManifest.new(readYaml(file: "manifests/$INPUT_MANIFEST"))
                         dockerBuild: {
                             build job: 'docker-build',
                             parameters: [
                                 string(name: 'DOCKER_BUILD_GIT_REPOSITORY', value: 'https://github.com/opensearch-project/opensearch-build'),
                                 string(name: 'DOCKER_BUILD_GIT_REPOSITORY_REFERENCE', value: 'main'),
-                                string(name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS', value: "id && pwd && cd docker/release && curl -sSL $URL_x64 -o opensearch-dashboards-x64.tgz && curl -sSL $URL_arm64 -o opensearch-dashboards-arm64.tgz && bash build-image-multi-arch.sh -v ${VERSION} -f ./dockerfiles/opensearch-dashboards.al2.dockerfile -p opensearch-dashboards -a 'x64,arm64' -r opensearchstaging/opensearch-dashboards -t 'opensearch-dashboards-x64.tgz,opensearch-dashboards-arm64.tgz' -n ${env.BUILD_NUMBER}"),
+                                string(name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS', value: [
+                                        'id',
+                                        'pwd',
+                                        'cd docker/release',
+                                        "curl -sSL ${inputManifest.getPublicDistPath(env.PUBLIC_ARTIFACT_URL, env.BUILD_NUMBER, 'linux','x64')} -o opensearch-dashboards-x64.tgz",
+                                        "curl -sSL ${inputManifest.getPublicDistPath(env.PUBLIC_ARTIFACT_URL, env.BUILD_NUMBER, 'linux','arm64')} -o opensearch-dashboards-arm64.tgz",
```

Comment:
Are these supposed to call `getPublicDistUrl` method?

Reply:
Good catch!